Code/Mantid/Framework/PythonInterface/plugins/algorithms/DNSMergeRuns.py

```diff
@@ -0,0 +1,285 @@
+            # we must sum up the values
+            for i in range(len(uX)):
+                idx = np.where(np.fabs(data_sorted[:, 0] - uX[i]) < 1e-4)
+                new_y = sum(data_sorted[idx][:, 1])
+                new_y_norm = sum(norm_sorted[idx][:, 1])
+                err = data_sorted[idx][:, 2]
+                err_norm = norm_sorted[idx][:, 2]
+                new_e = np.sqrt(np.dot(err, err))
+                new_e_norm = np.sqrt(np.dot(err_norm, err_norm))
+                arr.append([uX[i], new_y, new_e])
+                arr_norm.append([uX[i], new_y_norm, new_e_norm])
+            data = np.array(arr)
+            norm = np.array(arr_norm)
+
+        # define x axis
+        if self.xaxis == "2theta":
+            data[:, 0] = np.round(np.degrees(data[:, 0]), 2)
+            unitx = "Degrees"
+        elif self.xaxis == "|Q|":
+            data[:, 0] = np.fabs(4.0*np.pi*np.sin(0.5*data[:, 0])/self.wavelength)
+            unitx = "MomentumTransfer"
+        elif self.xaxis == "d-Spacing":
+            data[:, 0] = np.fabs(0.5*self.wavelength/np.sin(0.5*data[:, 0]))
+            unitx = "dSpacing"
+        else:
+            message = "The value for X axis " + self.xaxis + " is invalid! Cannot merge."
+            self.log().error(message)
+            raise RuntimeError(message)
+
+        data_sorted = data[np.argsort(data[:, 0])]
+        api.CreateWorkspace(dataX=data_sorted[:, 0], dataY=data_sorted[:, 1], dataE=data_sorted[:, 2],
+                            UnitX=unitx, OutputWorkspace=self.outws_name)
+        norm[:, 0] = data[:, 0]
+        norm_sorted = norm[np.argsort(norm[:, 0])]
+        api.CreateWorkspace(dataX=norm_sorted[:, 0], dataY=norm_sorted[:, 1], dataE=norm_sorted[:, 2],
+                            UnitX=unitx, OutputWorkspace=self.outws_name + '_NORM')
+        return
+
+    def PyExec(self):
+        # Input
+        normalize = self.getProperty("Normalize").value
+        self.workspace_names = self.getProperty("WorkspaceNames").value
+        self.outws_name = self.getProperty("OutputWorkspace").valueAsStr
+        self.xaxis = self.getProperty("HorizontalAxis").value
+
+        self.log().information("Workspaces to merge: %i" % (len(self.workspace_names)))
+        # check whether given workspaces can be merged
+        if not self._can_merge():
+            message = "Impossible to merge given workspaces."
+            raise RuntimeError(message)
+
+        self._merge_workspaces()
+        outws = api.AnalysisDataService.retrieve(self.outws_name)
+        api.CopyLogs(self.workspace_names[0], outws)
+        api.RemoveLogs(outws, self.sample_logs)
+
+        if normalize:
+            normws = api.AnalysisDataService.retrieve(self.outws_name + '_NORM')
+            api.Divide(LHSWorkspace=outws, RHSWorkspace=normws, OutputWorkspace=self.outws_name)
+            api.DeleteWorkspace(normws)
+
+        self.setProperty("OutputWorkspace", outws)
```

Comment:
The values in the output workspace don't seem to have units. I suggest setting them with setYUnit() and /or setYUnitLabel().